src/material/menu/menu-trigger.ts

```diff
@@ -103,7 +103,7 @@ export class MatMenuTrigger implements AfterContentInit, OnDestroy {
 
   // Tracking input type is necessary so it's possible to only auto-focus
   // the first item of the list when the menu is opened via the keyboard
-  _openedBy: 'mouse' | 'touch' | null = null;
+  _openedBy: 'mouse' | 'touch' | 'keyboard' | null = null;
```

Comment:
Should we just type this to `FocusOrigin`?

Reply:
It's not all the values of `FocusOrigin` though. I've changed it to `Exclude<FocusOrigin, 'program'>` instead.